services/web-app/utils/use-query-state.js

```diff
@@ -57,15 +28,35 @@ export const useQueryState = (
       return null
     }
 
-    const query = new URLSearchParams(window.location.search)
-    const queryValue = query.get(key)
-    const storageValue = saveToStorage ? localStorage.getItem(`${router.pathname}:${key}`) : null
+    const query = queryString.parseUrl(window.location.search, {
+      ...queryStringConfig,
+      parseNumbers,
+      parseBooleans
+    }).query
+
+    const storageValue = saveToStorage
+      ? localStorage.getItem(`${router.pathname}:${key}`)
+      : undefined
+
+    let val = query[key]
 
-    return queryValue !== null ? parse(queryValue) : storageValue
-  }, [key, parse, router.pathname, saveToStorage])
+    if (val !== undefined && validateValue && !validateValue(val)) {
+      val = undefined
+      delete query[key]
 
-  // Update the "state" when the router.query key changes
-  const value = useMemo(getValue, [getValue, router.query[key]])
+      // Change query state without rerendering page
+      history.replaceState(null, null, `?${queryString.stringify(query, queryStringConfig)}`)
```

Comment:
It looks like the goal of this logic is to remove invalid values from the query string. Is that desirable? Typically I'd expect to see the input data (query string) remain unaltered and have the application logic validate/sanitize it and then provide the valid values to the rest of the application.

Reply:
ok! I initially went with the "if this value is not valid for a key, remove it from the query" approach, but I can definitely leave the query string unaltered and ignore the invalid value instead

Reply:
Refactored to no longer delete invalid values and ignore them instead